### PR TITLE
book: cover UFCS in Syntax Index

### PR DESCRIPTION
**Pull request by durka**
r? @steveklabnik

### CODE REVIEW

File: src/doc/book/syntax-index.md [hunk 1]
```diff
@@ -117,6 +117,9 @@
 * `super::path`: path relative to the parent of the current module.  See [Crates and Modules (Re-exporting with `pub use`)].
 * `type::ident`: associated constants, functions, and types.  See [Associated Types].
 * `<type>::…`: associated item for a type which cannot be directly named (*e.g.* `<&T>::…`, `<[T]>::…`, *etc.*).  See [Associated Types].
+* `Trait::method(…)`: disambiguating a method call by naming the trait which defines it. See [Universal Function Call Syntax].
+* `Type::method(…)`: disambiguating a method call by naming the type for which it's defined. See [Universal Function Call Syntax].
+* `<Type as Trait>::method(…)`: disambiguating a method call by naming the trait _and_ type. See [Universal Function Call Syntax (Angle-bracket Form)].
```
Comment (durka) [inline src/doc/book/syntax-index.md]: Not sure if I should follow the (IMO odd) all-lowercase convention for these entries.

**Comment by durka** [comment]
Updated to follow the capitalization convention and cover the case of associated types (e.g. `<Type as Trait>::Assoc`).

**Comment by steveklabnik** [comment]
So, one thing here is that we've been meaning to rename UFCS for a long time, because it's a really bad name.

/cc @eddyb @rust-Lang/core

**Comment by durka** [comment]
How about something like "Fully Qualified Item" (by analogy with FQDN)?

**Comment by eddyb** [comment]
I agree on using "fully qualified", although there are "partially qualified" forms too.

@steveklabnik I don't think that worked. cc @rust-lang/core

**Comment by steveklabnik** [comment]
I think I'm gonna merge this, since it's not fair to block it on renaming UFCS. I just wanted to raise the issue.

@bors: r+ rollup

**Pull request merged**